libV2/cache/index.js

```diff
@@ -1,7 +1,27 @@
+const config = require('../config');
+
 const CacheClient = require('./client');
-const backends = require('./backend');
+const { MemoryCache, RedisCache } = require('./backend');
+
+let backend;
+switch (config.cacheBackend) {
+case 'memory':
+    backend = new MemoryCache();
+    break;
+
+case 'redis':
+    backend = new RedisCache(config.redis);
+    break;
+
+default:
+    throw Error(`Invalid cache backend ${config.cacheBackend}`);
+}
```

Comment:
What about using a map/object to go from cache backend string to cache object? Or instead of using cache backend just the presence of 'memory'/'redis' would use that object? Maybe something like:

```suggestion
const cacheTypes = {
  redis: RedisCache,
  memory: MemoryCache,
};
const backend = new cacheTypes[config.cacheBackend](config[config.cacheBackend] || {});
// this will throw an error if cache backend is undefined we may want to handle that...
```

Reply:
I tried a variation of your suggestion that I thinks a bit more readable.

Reply:
Agreed